src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -445,6 +479,42 @@ DropShards(Oid relationId, char *schemaName, char *relationName,
 }
 
 
+/*
+ * ExecuteDropShardPlacementCommandLocally prepares the task to DROP the given
+ * shard placement locally, executes that task if it can and returns true.
+ * If it cannot execute that task (see ShouldExecuteTasksLocally), returns false.
+ */
+static bool
+ExecuteDropShardPlacementCommandLocally(ShardPlacement *shardPlacement,
+										char *dropShardPlacementCommand)
+{
+	Assert(shardPlacement != NULL);
+
+	/* prepare the task for local execution */
+	Task *localPlacementTask = CitusMakeNode(Task);
+
+	localPlacementTask->jobId = INVALID_JOB_ID;
+	localPlacementTask->taskId = INVALID_TASK_ID;
+	localPlacementTask->taskType = DDL_TASK;
+	SetTaskQueryString(localPlacementTask, dropShardPlacementCommand);
+	localPlacementTask->dependentTaskList = NULL;
+	localPlacementTask->replicationModel = REPLICATION_MODEL_INVALID;
+	localPlacementTask->anchorShardId = shardPlacement->shardId;
+	localPlacementTask->taskPlacementList = list_make1(shardPlacement);
+
+	List *localPlacementTaskList = list_make1(localPlacementTask);
+
+	if (ShouldExecuteTasksLocally(localPlacementTaskList))
```

Comment:
I think the code flow would resemble other parts of local execution if we first decide `ShouldExecuteTasksLocally ` outside of the function and call `ExecuteDropShardPlacementCommandLocally ` without any further condition inside.

Reply:
makes sense